pkg/services/publicdashboards/database/database.go

```diff
@@ -149,118 +152,127 @@ func (d *PublicDashboardStoreImpl) FindByDashboardUid(ctx context.Context, orgId
 
 	return affectedRows, err
 }
 
-// ExistsEnabledByDashboardUid Responds true if there is an enabled public dashboard for a dashboard uid
-func (d *PublicDashboardStoreImpl) ExistsEnabledByDashboardUid(ctx context.Context, dashboardUid string) (bool, error) {
-	hasPublicDashboard := false
-	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
-		sql := "SELECT COUNT(*) FROM dashboard_public WHERE dashboard_uid=? AND is_enabled=true"
-
-		result, err := dbSession.SQL(sql, dashboardUid).Count()
+// Updates existing public dashboard
+func (d *PublicDashboardStoreImpl) Update(ctx context.Context, cmd SavePublicDashboardCommand) (int64, error) {
+	var affectedRows int64
+	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		timeSettingsJSON, err := json.Marshal(cmd.PublicDashboard.TimeSettings)
 		if err != nil {
 			return err
 		}
 
-		hasPublicDashboard = result > 0
-		return err
-	})
-
-	return hasPublicDashboard, err
-}
+		sqlResult, err := sess.Exec("UPDATE dashboard_public SET is_enabled = ?, annotations_enabled = ?, time_settings = ?, updated_by = ?, updated_at = ? WHERE uid = ?",
+			cmd.PublicDashboard.IsEnabled,
+			cmd.PublicDashboard.AnnotationsEnabled,
+			string(timeSettingsJSON),
+			cmd.PublicDashboard.UpdatedBy,
+			cmd.PublicDashboard.UpdatedAt.UTC().Format("2006-01-02 15:04:05"),
+			cmd.PublicDashboard.Uid)
 
-// ExistsEnabledByAccessToken Responds true if the accessToken exists and the public dashboard is enabled
-func (d *PublicDashboardStoreImpl) ExistsEnabledByAccessToken(ctx context.Context, accessToken string) (bool, error) {
-	hasPublicDashboard := false
-	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
-		sql := "SELECT COUNT(*) FROM dashboard_public WHERE access_token=? AND is_enabled=true"
+		if err != nil {
+			return err
+		}
 
-		result, err := dbSession.SQL(sql, accessToken).Count()
+		affectedRows, err = sqlResult.RowsAffected()
 		if err != nil {
 			return err
 		}
 
-		hasPublicDashboard = result > 0
-		return err
+		return nil
 	})
 
-	return hasPublicDashboard, err
+	return affectedRows, err
 }
 
-// GetOrgIdByAccessToken Returns the public dashboard OrgId if exists and is enabled.
-func (d *PublicDashboardStoreImpl) GetOrgIdByAccessToken(ctx context.Context, accessToken string) (int64, error) {
-	var orgId int64
-	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
-		sql := "SELECT org_id FROM dashboard_public WHERE access_token=? AND is_enabled=true"
+// Deletes a public dashboard
+func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, orgId int64, uid string) (int64, error) {
+	dashboard := &PublicDashboard{OrgId: orgId, Uid: uid}
+	var affectedRows int64
+	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		var err error
+		affectedRows, err = sess.Delete(dashboard)
 
-		_, err := dbSession.SQL(sql, accessToken).Get(&orgId)
 		if err != nil {
 			return err
 		}
-
-		return err
+		return nil
```

Comment:
does it make any sense? if err is not nil, we return err. Otherwise, (if it's nil), we return nil. 
Why not just return err?